src/Body/Fields/File/File.js

```diff
@@ -15,51 +15,81 @@ const File = ({
   },
   dropbox,
   validation,
+  fileList,
+  setFileList,
   ...props
 }) => {
   const fileInputRef = useRef(null)
-  const [fileList, setFileList] = useState([])
-  const [cancelRequestId, setCancelRequestId] = useState('')
-  const { neededValues } = useField([id])
+
   useEffect(() => {
-    ABORT_REQUEST_CONTROLLERS.get(cancelRequestId)?.abort()
-    window._handleChange({ id, value: fileList.length === 0 ? null : fileList })
-  }, [fileList, cancelRequestId])
+    window._handleChange({ id, value: fileList[id]?.length === 0 ? null : fileList[id] })
+  }, [fileList])
+
 
   const handleCallback = (fileId, status, responce, fileName) => {
 
     if (status === 200) {
-      setFileList((prev) => prev.map((file) => {
-        if (fileId === file.id) file.responce = responce
-        return file
-      }))
+      setFileList((prev) => {
+        let fileResponce = prev[id].map((file) => {
+          if (file.id === fileId) {
+            file.responce = responce
+          }
+          return file
+        })
+        return {
+         ...prev,
+          [id]:[...fileResponce]
+       }
+     })
     }
 
     if (responce?.message === 'canceled') {
       console.log(`You have canceled ${fileName} file upload`)
     } else if (status === 0) {
-      const incorrectFile = fileList.find((item) => item.id === fileId)
+      const incorrectFile = fileList[id].find((item) => item.id === fileId)
       console.log(
-        `We are unable to upload your file named ${incorrectFile.name}. Please if it’s possible try to rename it, otherwise contact us.`
+        `We are unable to upload your file named ${incorrectFile?.name}. Please if it’s possible try to rename it, otherwise contact us.`
       )
     }
   }
 
   const handleLoading = (fileId, percentage) => {
-
-    setFileList((prev) => {
-      return prev.map((file) => {
-        if (fileId === file.id) file.percentage = percentage
-        return file
+    if ((percentage === 100 || percentage === 0)) {
+      
+      setFileList((prev) => {
+        let filePrecentage = prev[id].map((file) => {
+          if (file.id === fileId) {
+            file.percentage = percentage
+          }
+          return file
+        })
+        return {
+          ...prev,
+          [id]: [...filePrecentage],
+        }
```

Comment:
If we are using only percentage === 0 which is start uploading and percentage === 100 which is file successfully uploaded. Why we don't use acceptFiles function which is file started to upload and  handleCallback which is file finished uploading ?